__tests__/omnibox-directions.test.tsx

```diff
@@ -0,0 +1,274 @@
+        startLocation={mockPOIs[0]}
+        endLocation={mockPOIs[1]}
+        setEndLocationFocused={mockSetIsDestinationFocused}
+        currentLocation={null}
+        setEndLocation={null}
+        setStartLocation={null}
+        queryText={null}
+        endLocationFocused={null}
+        setTravelState={null}
+      />
+    );
+
+    const textInput = wrapper.find({ testID: "searchInputInitialLocation" });
+    textInput.simulate("focus");
+    expect(mockSetIsDestinationFocused).toHaveBeenCalledTimes(1);
+    expect(mockSetIsDestinationFocused.mock.calls[0][0]).toBe(false);
+  });
+
+  it("should call a function when endLocation location textInput text is changed", () => {
+    const mockQueryText = jest.fn();
+
+    const wrapper = shallow(
+      <OmniboxDirections
+        startLocation={mockPOIs[0]}
+        endLocation={mockPOIs[1]}
+        queryText={mockQueryText}
+        currentLocation={null}
+        setEndLocation={null}
+        setStartLocation={null}
+        setEndLocationFocused={null}
+        endLocationFocused={null}
+        setTravelState={null}
+      />
+    );
+
+    const textInput = wrapper.find({
+      testID: "searchInputDestinationLocation"
+    });
+    textInput.simulate("changeText", "H805");
+    expect(mockQueryText).toHaveBeenCalledTimes(1);
+    expect(mockQueryText.mock.calls[0][0]).toBe("H805");
+  });
+
+  it("should call the provided functin when endLocation location textInput is focused", () => {
+    const mockSetIsDestinationFocused = jest.fn();
+
+    const wrapper = shallow(
+      <OmniboxDirections
+        startLocation={mockPOIs[0]}
+        endLocation={mockPOIs[1]}
+        setEndLocationFocused={mockSetIsDestinationFocused}
+        currentLocation={null}
+        setEndLocation={null}
+        setStartLocation={null}
+        queryText={null}
+        endLocationFocused={null}
+        setTravelState={null}
+      />
+    );
+
+    const textInput = wrapper.find({
+      testID: "searchInputDestinationLocation"
+    });
+    textInput.simulate("focus");
+    expect(mockSetIsDestinationFocused).toHaveBeenCalledTimes(1);
+    expect(mockSetIsDestinationFocused.mock.calls[0][0]).toBe(true);
+  });
+
+  it("should call setshowTimePicker() when pressed", () => {
+    const mockSetshowTimePicker = jest.fn();
+
+    const useStateSpy = jest.spyOn(React, "useState");
+    // @ts-ignore
+    useStateSpy.mockImplementation(init => [init, mockSetshowTimePicker]);
+
+    const wrapper = shallow(
+      <OmniboxDirections
+        startLocation={mockPOIs[0]}
+        endLocation={mockPOIs[1]}
+        currentLocation={null}
+        setEndLocation={null}
+        setStartLocation={null}
+        queryText={null}
+        setEndLocationFocused={null}
+        endLocationFocused={null}
+        setTravelState={null}
+      />
+    );
+
+    wrapper
+      .find({ testID: "timePickerButton" })
+      .props()
+      .onPress();
+
+    expect(mockSetshowTimePicker).toHaveBeenCalledTimes(1);
+  });
```

Comment:
This cannot be done since it needs an initial value to set and null can't be passed, explained here : https://dev.to/theactualgivens/testing-react-hook-state-changes-2oga